it/src/main/java/org/corfudb/universe/universe/vm/ApplianceManager.java

```diff
@@ -0,0 +1,220 @@
+                .keySet()
+                .stream()
+                .map(this::deployVmAsync)
+                .collect(Collectors.toList());
+
+        asyncDeployment
+                .stream()
+                .map(CompletableFuture::join)
+                .forEach(vm -> {
+                    universeParams.updateIpAddress(vm.getName(), vm.getGuest().getIpAddress());
+                    deployment.put(vm.getName(), vm);
+                });
+
+        log.info("The deployed VMs are: {}", universeParams.getVmIpAddresses());
+
+        vms.putAll(deployment);
+    }
+
+    private CompletableFuture<VirtualMachine> deployVmAsync(String vmName) {
+        log.info("Deploy vm asynchronously: {}", vmName);
+        return CompletableFuture.supplyAsync(() -> deployVm(vmName), executor);
+    }
+
+    /**
+     * Deploy and power on a VM appliance in vSphere.
+     *
+     * @return VirtualMachine instance
+     */
+    private VirtualMachine deployVm(String vmName) {
+        log.info("Deploy VM: {}", vmName);
+
+        VirtualMachine vm;
+        try {
+            // Connect to vSphere server using VIJAVA
+            ServiceInstance si = new ServiceInstance(
+                    new URL(universeParams.getVSphereUrl()),
+                    universeParams.getVSphereUsername(),
+                    universeParams.getVSpherePassword(),
+                    true
+            );
+
+            InventoryNavigator inventoryNavigator = new InventoryNavigator(si.getRootFolder());
+
+            // First check if a VM with this name already exists or not
+            vm = (VirtualMachine) inventoryNavigator.searchManagedEntity(
+                    ManagedEntityType.VIRTUAL_MACHINE.typeName, vmName);
+            if (vm != null) {
+                // If the VM already exists, ensure the power state is 'on' before return the VM
+                if (vm.getSummary().runtime.powerState == VirtualMachinePowerState.poweredOff) {
+                    log.info(vmName + " already exists, but found in 'poweredOff' state, powering it on...");
+                    Task task = vm.powerOnVM_Task((HostSystem) inventoryNavigator.searchManagedEntity(
+                            "HostSystem", universeParams.getVSphereHost()));
+                    task.waitForTask();
+                }
+            } else {
+                // Find the template machine in the inventory
+                VirtualMachine vmTemplate = (VirtualMachine) inventoryNavigator.searchManagedEntity(
+                        ManagedEntityType.VIRTUAL_MACHINE.typeName,
+                        universeParams.getTemplateVMName()
+                );
+
+                log.info("Deploying the VM {} via vSphere {}...", vmName, universeParams.getVSphereUrl());
+
+                // Create customization for cloning process
+                VirtualMachineCloneSpec cloneSpec = createLinuxCustomization(vmName);
+                try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)